From f4b157ad09c71afa81634b80c496ab9a99d42fa3 Mon Sep 17 00:00:00 2001
From: Virendra Kakade <virendra.kakade@ni.com>
Date: Thu, 30 Nov 2023 13:55:13 -0600
Subject: [PATCH] board: ni: fix PS dram configuration

We were using settings for an incorrect DRAM part. Use the correct
settings.
These values were generated in Vivado 2019.1 and the psu_init_gpl.c was
compared and settings changes added.

Signed-off-by: Virendra Kakade <virendra.kakade@ni.com>
---
 board/ni/zynqmp/x410-rev5/psu_init_gpl.c | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/board/ni/zynqmp/x410-rev5/psu_init_gpl.c b/board/ni/zynqmp/x410-rev5/psu_init_gpl.c
index 39cac33519..412eaebe6e 100644
--- a/board/ni/zynqmp/x410-rev5/psu_init_gpl.c
+++ b/board/ni/zynqmp/x410-rev5/psu_init_gpl.c
@@ -117,7 +117,7 @@ static unsigned long psu_ddr_init_data(void)
 	psu_mask_write(0xFD0700D0, 0xC3FF0FFFU, 0x00020126U);
 	psu_mask_write(0xFD0700D4, 0x01FF7F0FU, 0x00020000U);
 	psu_mask_write(0xFD0700D8, 0x0000FF0FU, 0x00002705U);
-	psu_mask_write(0xFD0700DC, 0xFFFFFFFFU, 0x09340301U);
+	psu_mask_write(0xFD0700DC, 0xFFFFFFFFU, 0x09400301U);
 	psu_mask_write(0xFD0700E0, 0xFFFFFFFFU, 0x00D80200U);
 	psu_mask_write(0xFD0700E4, 0x00FF03FFU, 0x00210004U);
 	psu_mask_write(0xFD0700E8, 0xFFFFFFFFU, 0x000C06C0U);
@@ -126,9 +126,9 @@ static unsigned long psu_ddr_init_data(void)
 	psu_mask_write(0xFD0700F4, 0x00000FFFU, 0x0000066FU);
 	psu_mask_write(0xFD070100, 0x7F3F7F3FU, 0x11122813U);
 	psu_mask_write(0xFD070104, 0x001F1F7FU, 0x0004041CU);
-	psu_mask_write(0xFD070108, 0x3F3F3F3FU, 0x0608070DU);
+	psu_mask_write(0xFD070108, 0x3F3F3F3FU, 0x0609080DU);
 	psu_mask_write(0xFD07010C, 0x3FF3F3FFU, 0x0050400CU);
-	psu_mask_write(0xFD070110, 0x1F0F0F1FU, 0x08030409U);
+	psu_mask_write(0xFD070110, 0x1F0F0F1FU, 0x0903040AU);
 	psu_mask_write(0xFD070114, 0x0F0F3F1FU, 0x06060403U);
 	psu_mask_write(0xFD070118, 0x0F0F000FU, 0x01010004U);
 	psu_mask_write(0xFD07011C, 0x00000F0FU, 0x00000606U);
@@ -138,14 +138,14 @@ static unsigned long psu_ddr_init_data(void)
 	psu_mask_write(0xFD070130, 0x00030F1FU, 0x00020608U);
 	psu_mask_write(0xFD070180, 0xF7FF03FFU, 0x81000040U);
 	psu_mask_write(0xFD070184, 0x3FFFFFFFU, 0x0201C9C2U);
-	psu_mask_write(0xFD070190, 0x1FBFBF3FU, 0x048C8209U);
+	psu_mask_write(0xFD070190, 0x1FBFBF3FU, 0x048E8209U);
 	psu_mask_write(0xFD070194, 0xF31F0F0FU, 0x00030304U);
 	psu_mask_write(0xFD070198, 0x0FF1F1F1U, 0x07000101U);
 	psu_mask_write(0xFD07019C, 0x000000F1U, 0x00000021U);
 	psu_mask_write(0xFD0701A0, 0xC3FF03FFU, 0x00400003U);
 	psu_mask_write(0xFD0701A4, 0x00FF00FFU, 0x00C800FFU);
 	psu_mask_write(0xFD0701B0, 0x00000007U, 0x00000000U);
-	psu_mask_write(0xFD0701B4, 0x00003F3FU, 0x00000A07U);
+	psu_mask_write(0xFD0701B4, 0x00003F3FU, 0x00000C07U);
 	psu_mask_write(0xFD0701C0, 0x00000007U, 0x00000001U);
 	psu_mask_write(0xFD070200, 0x0000001FU, 0x0000001FU);
 	psu_mask_write(0xFD070204, 0x001F1F1FU, 0x001F0909U);
@@ -159,7 +159,7 @@ static unsigned long psu_ddr_init_data(void)
 	psu_mask_write(0xFD070224, 0x0F0F0F0FU, 0x07070707U);
 	psu_mask_write(0xFD070228, 0x0F0F0F0FU, 0x07070707U);
 	psu_mask_write(0xFD07022C, 0x0000000FU, 0x00000007U);
-	psu_mask_write(0xFD070240, 0x0F1F0F7CU, 0x0600060CU);
+	psu_mask_write(0xFD070240, 0x0F1F0F7CU, 0x06000614U);
 	psu_mask_write(0xFD070244, 0x00003333U, 0x00000001U);
 	psu_mask_write(0xFD070250, 0x7FFF3F07U, 0x01002001U);
 	psu_mask_write(0xFD070264, 0xFF00FFFFU, 0x08000040U);
@@ -227,18 +227,18 @@ static unsigned long psu_ddr_init_data(void)
 	psu_mask_write(0xFD0800C0, 0xFFFFFFFFU, 0x00000000U);
 	psu_mask_write(0xFD0800C4, 0xFFFFFFFFU, 0x000000E0U);
 	psu_mask_write(0xFD080100, 0xFFFFFFFFU, 0x0800040CU);
-	psu_mask_write(0xFD080110, 0xFFFFFFFFU, 0x08261009U);
+	psu_mask_write(0xFD080110, 0xFFFFFFFFU, 0x08261209U);
 	psu_mask_write(0xFD080114, 0xFFFFFFFFU, 0x28240008U);
 	psu_mask_write(0xFD080118, 0xFFFFFFFFU, 0x000F0300U);
 	psu_mask_write(0xFD08011C, 0xFFFFFFFFU, 0x83000800U);
 	psu_mask_write(0xFD080120, 0xFFFFFFFFU, 0x01A42B08U);
-	psu_mask_write(0xFD080124, 0xFFFFFFFFU, 0x00371009U);
-	psu_mask_write(0xFD080128, 0xFFFFFFFFU, 0x00000C10U);
+	psu_mask_write(0xFD080124, 0xFFFFFFFFU, 0x00381209U);
+	psu_mask_write(0xFD080128, 0xFFFFFFFFU, 0x00000C12U);
 	psu_mask_write(0xFD080140, 0xFFFFFFFFU, 0x08400020U);
 	psu_mask_write(0xFD080144, 0xFFFFFFFFU, 0x00000C80U);
 	psu_mask_write(0xFD080150, 0xFFFFFFFFU, 0x00000000U);
 	psu_mask_write(0xFD080154, 0xFFFFFFFFU, 0x00000300U);
-	psu_mask_write(0xFD080180, 0xFFFFFFFFU, 0x00000834U);
+	psu_mask_write(0xFD080180, 0xFFFFFFFFU, 0x00000840U);
 	psu_mask_write(0xFD080184, 0xFFFFFFFFU, 0x00000301U);
 	psu_mask_write(0xFD080188, 0xFFFFFFFFU, 0x000000D8U);
 	psu_mask_write(0xFD08018C, 0xFFFFFFFFU, 0x00000200U);
-- 
2.25.1

